app/serializers/basic_user_serializer.rb

```diff
@@ -46,4 +46,12 @@ def tags_with_notification_level(lookup_level)
   def tag_user_notification_levels
     @tag_user_notification_levels ||= TagUser.notification_levels_for(user)
   end
+
+  def external_id
+    user.single_sign_on_record&.external_id
+  end
+
+  def include_external_id?
+    user.respond_to?(:single_sign_on_record) and SiteSetting.enable_discourse_connect_external_id_serializers?
```

Comment:
We try to avoid using `and` and `or` and instead use `&&` and `||` which have different precedence behavior

```suggestion
    user.respond_to?(:single_sign_on_record) && SiteSetting.enable_discourse_connect_external_id_serializers?
```